lib/dor/models/describable.rb

```diff
@@ -41,31 +41,27 @@ def build_descMetadata_datastream(ds)
 
     # Generates Dublin Core from the MODS in the descMetadata datastream using the LoC mods2dc stylesheet
     #    Should not be used for the Fedora DC datastream
-    # @raise [Exception] Raises an Exception if the generated DC is empty or has no children
+    # @raise [CrosswalkError] Raises an Exception if the generated DC is empty or has no children
+    # @return [Nokogiri::Doc] the DublinCore XML document object
     def generate_dublin_core
-      format = metadata_format
-      if format.nil?
-        raise CrosswalkError, "Unknown descMetadata namespace: #{metadata_namespace.inspect}"
-      end
-      xslt = Nokogiri::XSLT(File.new(File.expand_path(File.dirname(__FILE__) + "/#{format}2dc.xslt")) )
+      raise CrosswalkError, "Unknown descMetadata namespace: #{metadata_namespace.inspect}" if metadata_format.nil?
+      xslt = Nokogiri::XSLT(File.new(File.expand_path(File.dirname(__FILE__) + "/#{metadata_format}2dc.xslt")) )
```

Comment:
We could put these xslt's into a cache rather than reparsing them everytime. There's really only 1 of these xslts `mods2dc.xslt`. It's overly flexible here.